### PR TITLE
add tailwindcss

### PR DESCRIPTION
**Pull request by dancancode85**

### CODE REVIEW

File: src/components/ViewArticles.jsx [hunk 1]
```diff
@@ -1,9 +1,33 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
 
 function ViewArticles() {
+  const [articles, setArticles] = useState([]);
+
+  useEffect(() => {
+    axios.get('http://localhost:9090/api/articles')
```
Comment (saimakhanom) [inline src/components/ViewArticles.jsx]: this should be extracted out to its own file in a utils folder

File: src/components/ViewArticles.jsx [hunk 2]
```diff
@@ -1,9 +1,33 @@
...
+      .then(res => {
+        setArticles(res.data.articles);
+        console.log(res.data.articles);
```
Comment (saimakhanom) [inline src/components/ViewArticles.jsx]: remove logs

File: src/components/ViewArticles.jsx [hunk 3]
```diff
@@ -1,9 +1,33 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
 
 function ViewArticles() {
+  const [articles, setArticles] = useState([]);
```
Comment (saimakhanom) [inline src/components/ViewArticles.jsx:5]: we also need a loading state to provide visual feedback to the user on slow load

File: src/components/ViewArticles.jsx [hunk 4]
```diff
@@ -1,9 +1,33 @@
...
+  }, []);
+
   return (
-    <div>ViewArticles</div>
-  )
+    <div>
+      <h1>View Articles</h1>
+      {articles.map(article => (
+    
+        <div key={article.article_id}>
```
Comment (saimakhanom) [inline src/components/ViewArticles.jsx]: for the article previews we want to see a bit more information before we click on them and read the full article - like the number of votes and comments it has and the date it was posted (in a human readble format). We dont want to see the article body until we click through to a single article page

File: src/components/utils/api.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios';
+
+const api = axios.create({
+  baseURL: 'http://localhost:9090/api',
+});
+
+export const getAllArticles = () => {
```
Comment (saimakhanom) [inline src/components/utils/api.js:7]: this should replace the axios.get in view articles

Comment (dancancode85) [reply src/components/utils/api.js:7]: Done!

File: src/components/ViewArticles.jsx (hunk 4 above)
Comment (dancancode85) [reply src/components/ViewArticles.jsx]: Done!

File: src/components/ViewArticles.jsx (hunk 2 above)
Comment (dancancode85) [reply src/components/ViewArticles.jsx]: Done!

File: src/components/ViewArticles.jsx (hunk 1 above)
Comment (dancancode85) [reply src/components/ViewArticles.jsx]: Done!

File: src/components/ViewArticles.jsx (hunk 3 above)
Comment (dancancode85) [reply src/components/ViewArticles.jsx:5]: Done!!

File: src/components/ViewArticles.jsx [hunk 6]
```diff
@@ -3,33 +3,57 @@ import React, { useState, useEffect } from 'react';
...
-          <img src={article.article_img_url} alt={`A picture of ${article.title}`} />
-          <h2>{article.title}</h2>
-       
-          <p>Written by {article.author}</p>
-
-          <p>Posted on {new Date(article.created_at).toLocaleDateString('en-GB', { day: 'numeric', month: 'long', year: 'numeric' })}</p>          
-          <p>{article.votes} votes</p>
-          <p>{article.comment_count} comments</p>
-          Category: <a href="">coding</a>
+    <div className="loading-spinner">
```
Comment (saimakhanom) [inline src/components/ViewArticles.jsx:20]: try making this a resuable component - easier than having to copy and paste this every time you want loading functionality

**Pull request merged**